Prox/Prox/Data/PlacesController.swift

```diff
@@ -139,6 +129,28 @@ class PlacesProvider {
         }
     }
 
+    func filterPlaces(filters: [PlaceFilter]) -> [Place] {
+        let enabledCategories = Set(filters.filter { $0.enabled }.map { $0.categories }.reduce([], +))
+        let toRoots = CategoriesUtil.categoryToRootsMap
+
+        return allPlaces.filter { place in
```

Comment:
nit: should we lock for access to allPlaces?

Note: this is called from `updateDisplayedPlaces` which expects the caller to lock so this method would also need to be locked by the called (but it's public!).